temporian/implementation/numpy/data/event.py

```diff
@@ -199,144 +262,190 @@ def from_dataframe(
 
         # Representation of the "data" field
         with np.printoptions(precision=4, threshold=20):
             data_repr = []
-            for idx, (k, v) in enumerate(self.data.items()):
-                if idx > MAX_NUM_PRINTED_INDEX:
+            for i, (index_key, index_data) in enumerate(self._data.items()):
+                if i > MAX_NUM_PRINTED_INDEX:
                     data_repr.append("...")
                     break
-                data_repr.append(f"{k}:\n{string.indent(repr_features(v))}")
+                data_repr.append(
+                    f"{index_key}:"
+                    f" {index_data.timestamps}\n{string.indent(repr_features(index_data.features))}"
+                )
             data_repr = string.indent("\n".join(data_repr))
 
-        # Representation of the "sampling" field
-        sampling_repr = string.indent(self.sampling.__repr__())
-
         return (
-            f"data ({len(self.data)}):\n{data_repr}\nsampling:\n{sampling_repr}"
+            "data:"
+            f"\n\t\tindex_names={self.index_names}"
+            f"\n\t\tfeature_names={self.feature_names}"
+            f"\n{data_repr}"
         )
 
+    def __getitem__(self, index: Tuple) -> IndexData:
+        return self.data[index]
+
+    def __setitem__(self, index: Tuple, value: IndexData) -> None:
+        self.data[index] = value
+
     def __eq__(self, __o: object) -> bool:
         if not isinstance(__o, NumpyEvent):
             return False
 
-        # Check equal sampling and index values
-        if self.sampling != __o.sampling:
+        # check same features
+        if self._feature_names != __o.feature_names:
             return False
 
-        # Check same features
-        if self.feature_names() != __o.feature_names():
+        # check same index
+        if self._index_names != __o.index_names:
             return False
 
-        # Check each feature is equal in each index
-        for index in self.data.keys():
-            # Check both feature list are equal
-            if self.data[index] != __o.data[index]:
-                return False
+        # check unix timestamp
+        if self._is_unix_timestamp != __o.is_unix_timestamp:
+            return False
 
-        return True
+        # check same data
+        if len(self._data) != len(__o.data):
+            return False
+
+        for index_key, index_data_self in self._data.items():
+            if index_key not in __o.data:
+                return False
 
-    def index(self) -> Sequence[Any]:
-        """Sequence of available indexes."""
+            # check same features
+            index_data_other = __o[index_key]
+            for feature_self, feature_other in zip(
+                index_data_self.features, index_data_other.features
+            ):
+                if feature_self.dtype.type != feature_other.dtype.type:
+                    return False
+
+                # check if the array has a float dtype. If so, compare with
```

Comment:
do we want event with different dtypes but same values to be the same?

Reply:
I don't think so